### PR TITLE
Fixes file uri (NullPointerExceptoin

### PR DESCRIPTION
**Pull request by jarl-dk**
This solves the issue mentioned in https://github.com/xmlresolver/xmlresolver/issues/174

This supersedes https://github.com/xmlresolver/xmlresolver/pull/176

This includes a test

### CODE REVIEW

File: src/main/java/org/xmlresolver/ResourceAccess.java [hunk 1]
```diff
@@ -216,25 +216,18 @@ private static ResourceResponse getNetResource(ResourceRequest request, URI reso
...
+        }
 
-            if (URIUtils.forbidAccess(accessList, requestURI.toString(), mergeHttps)) {
-                if (request.isResolvingAsEntity()) {
-                    logger.log(AbstractLogger.REQUEST, "resolveEntity, access denied: " + requestURI);
-                } else {
-                    logger.log(AbstractLogger.REQUEST, "resolveURI, access denied: " + requestURI);
-                }
-                return new ResourceResponse(request, true);
+        if (URIUtils.forbidAccess(accessList.concat(",file"), requestURI.toString(), mergeHttps)) {
```
Comment (jarl-dk) [inline src/main/java/org/xmlresolver/ResourceAccess.java:224]: Notice that it is necessary to add "file" protocol.

**Comment by jarl-dk** [comment]
@ndw, Will you comment on this?

**Comment by ndw** [comment]
Apologies. I have been distracted. I'll review (and most likely merge) this today or this weekend.

**Comment by jarl-dk** [comment]
Please also consider the other PRs.

**Comment by jarl-dk** [comment]
> Apologies. I have been distracted. I'll review (and most likely merge) this today or this weekend.

I suggest you make a more thorough review of the implementation. The real value in the PR is the test, not the implementation. There are many different implementations (various design ideas) to make the test green. Consider you design intention. But then again, the implementation can always be refactored later. I am an advocate of _Red, Green, Refactor_.

**Pull request closed without merging**